app/src/lib/stats/stats-store.ts

```diff
@@ -989,6 +998,62 @@ export class StatsStore implements IStatsStore {
     }))
   }
 
+  /** Record when the user views a stash entry after checking out a branch */
+  public async recordStashViewedAfterCheckout(): Promise<void> {
```

Comment:
:art: as none of these new functions use `await`, the `async` attribute can be dropped:

```suggestion
  public recordStashViewedAfterCheckout(): Promise<void> {
```

Reply:
@iAmWillShepherd just an FYI - there's another seven function in here that can use the same treatment - I just didn't explicitly go through them all

Reply:
I see there are actually more than 7 that this can be done to. I'll open up a follow-up PR where I fix them all.

Reply:
Feedback addressed in https://github.com/desktop/desktop/pull/7640.

Reply:
#7640 is fine - I just wanted to avoid adding more code that includes the unnecessary keyword

Reply:
Jkk